platform/ios/MGLMapboxEvents.m

```diff
@@ -483,22 +565,13 @@ - (void) startTimer {
 
 // Can be called from any thread.
 //
-- (NSString *) getUserAgent {
-    if (self.appName != nil && self.appVersion != nil && self.appBuildNumber != nil && ([self.userAgent rangeOfString:self.appName].location == NSNotFound)) {
-        self.userAgent = [NSString stringWithFormat:@"%@/%@/%@ %@", self.appName, self.appVersion, self.appBuildNumber, self.userAgent];
-    }
-    return self.userAgent;
-}
-
-// Can be called from any thread.
-//
-- (NSString *) formatDate:(NSDate *)date {
-    return [self.rfc3339DateFormatter stringFromDate:date];
+- (NSString *) userAgent {
+    return [NSString stringWithFormat:@"%@/%@/%@ %@", self.appName, self.appVersion, self.appBuildNumber, MGLMapboxEventsUserAgent];
```

Comment:
These properties are now [set unconditionally in `-init`](https://github.com/mapbox/mapbox-gl-native/blob/1ec5-metrics-opt-way-out/platform/ios/MGLMapboxEvents.m#L214-216) based on Info.plist fields that are required for the app to launch. Even if they’re `nil` for some reason, you’d wind up with a user agent string that starts with “(null)/(null)/(null)”. The old code would simply return `nil` and crash when [inserted](https://github.com/mapbox/mapbox-gl-native/blob/b88b917cd46f58b7acf6f97a26a9a7ec6e1a257f/platform/ios/MGLMapboxEvents.m#L445) into the HTTP request.